### PR TITLE
[PixArt-Alpha] fix mask feature condition.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

https://github.com/huggingface/diffusers/pull/5677 actually broke the default behavior and was leading to crappy results. This PR fixes that. 

Sincere apologies for rushing here. 

Cc: @lawrence-cj

### CODE REVIEW

**Review by lawrence-cj — APPROVED** [review]
Checked, fixed the mask feature bug

File: tests/pipelines/pixart/test_pixart.py [hunk 1]
```diff
@@ -120,7 +120,6 @@ def test_save_load_optional_components(self):
             "generator": generator,
             "num_inference_steps": num_inference_steps,
             "output_type": output_type,
-            "mask_feature": False,
```
Comment (sayakpaul) [inline tests/pipelines/pixart/test_pixart.py:123]: @patrickvonplaten with the current code changes, even if the `mask_feature` argument is set to True as long as the initial embeds are provided, this code path is not going to get hit:

https://github.com/huggingface/diffusers/blob/69996938cf7b41d4a69aea7aa6ab08a9b76a115d/src/diffusers/pipelines/pixart_alpha/pipeline_pixart_alpha.py#L256

Earlier it wasn't getting hit improperly. Let me know if that makes sense.

Comment (patrickvonplaten) [reply tests/pipelines/pixart/test_pixart.py:123]: ok!

**Review by patrickvonplaten — APPROVED** [review]

**Comment by sayakpaul** [comment]
Failing test is unrelated.

**Pull request merged**